crds/tests/test_config.py

```diff
@@ -23,6 +27,49 @@
 TEST_MAPPATH = os.path.join(CRDS_TESTING_CACHE, "mappings")
 TEST_TEMP_DIR = tempfile.mkdtemp(prefix='crds-test-')
 
+
+def generate_test_data():
+    import asdf
+    from roman_datamodels import datamodels as rdd
+    from roman_datamodels.testing import factories
+
+    reftype = {
+        'flat': factories.create_flat_ref,
+        'dark': factories.create_dark_ref,
+        'gain': factories.create_gain_ref,
+        'mask': factories.create_mask_ref,
+        'readnoise': factories.create_readnoise_ref
+    }
+
+    test_files = {
+        'roman_wfi16_grism_flat_small.asdf': 'flat',
+        'roman_wfi16_grism_flat_invalid_tpn.asdf': 'flat',
```

Comment:
Given the name, I suspect that this file needs to be invalid in a specific way, but this code generates a valid file

Reply:
no doubt, but I am not generating any specific use cases until the base cases pass then I will update the autogen function to address it. I am sure the test will fail with a helpful error.